core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -464,11 +517,14 @@ class PartKeyLuceneIndex(ref: DatasetRef,
       logger.warn(s"Could not find in Lucene startTime for partId=$partId in dataset=$ref. Using " +
         s"$startTime instead.", new IllegalStateException()) // assume this time series started retention period ago
     }
-    val updatedDoc = makeDocument(partKeyOnHeapBytes, partKeyBytesRefOffset, partKeyNumBytes,
+    makeDocument(partKeyOnHeapBytes, partKeyBytesRefOffset, partKeyNumBytes,
       partId, startTime, endTime)
+
+    val doc = luceneDocument.get()
```

Comment:
Any reason why the return value of ``makeDocument`` is ignored and retrieved from ``ThreadLocal`` again?

Reply:
yeah its not consistent too, at some places return value is still used.

Reply:
I have now removed assignment of makeDocument return value, which returns the actual lucene document. The problem is we need the FacetsConfig object too.

The thread local (`ReusableDocument` object) stores both the lucene `Document` and the `FacetsConfig` object which are both needed to build the final document that will be added to index.